tests/skykeys_integration_test.go

```diff
@@ -0,0 +1,119 @@
+package tests
+
+import (
+	"reflect"
+	"testing"
+
+	skynet "github.com/NebulousLabs/go-skynet"
+	"gopkg.in/h2non/gock.v1"
+)
+
+// TestAddSkykey tests adding a skykey.
+func TestAddSkykey(t *testing.T) {
+	defer gock.Off()
+
+	const skykey = "skykey:BAAAAAAAAABrZXkxAAAAAAAAAAQgAAAAAAAAADiObVg49-0juJ8udAx4qMW-TEHgDxfjA0fjJSNBuJ4a"
+
+	opts := skynet.DefaultAddSkykeyOptions
+	gock.New(skynet.DefaultPortalURL).
+		Post(opts.PortalAddSkykeyPath).
+		MatchParam("skykey", skykey).
+		Reply(200)
+
+	err := skynet.AddSkykey(skykey, opts)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Verify we don't have pending mocks.
+	if !gock.IsDone() {
+		t.Fatal("test finished with pending mocks")
+	}
+}
+
+// TestCreateSkykey tests creating a skykey.
+func TestCreateSkykey(t *testing.T) {
+	defer gock.Off()
+
+	const skykey = "skykey:BAAAAAAAAABrZXkxAAAAAAAAAAQgAAAAAAAAADiObVg49-0juJ8udAx4qMW-TEHgDxfjA0fjJSNBuJ4a"
+	const name = "testcreateskykey"
+	const id = "pJAPPfWkWXpss3BvMDCJCw=="
+	const skykeyType = "private-id"
+
+	opts := skynet.DefaultCreateSkykeyOptions
+	gock.New(skynet.DefaultPortalURL).
+		Post(opts.PortalCreateSkykeyPath).
+		MatchParam("name", name).
+		MatchParam("type", skykeyType).
+		Reply(200).
+		JSON(skynet.Skykey{Skykey: skykey, Name: name, ID: id, Type: skykeyType})
+
+	skykey2, err := skynet.CreateSkykey(name, skykeyType, opts)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	expectedSkykey := skynet.Skykey{
+		Skykey: skykey,
+		Name:   name,
+		ID:     id,
+		Type:   skykeyType,
+	}
+	if !reflect.DeepEqual(expectedSkykey, skykey2) {
+		t.Fatalf("expected skykey %v, got %v", expectedSkykey, skykey2)
+	}
+
+	// Verify we don't have pending mocks.
+	if !gock.IsDone() {
+		t.Fatal("test finished with pending mocks")
+	}
+}
+
+// TestGetSkykey tests getting a skykey by name and by ID.
+func TestGetSkykey(t *testing.T) {
+	defer gock.Off()
+
+	const name = "testcreateskykey"
+	const id = "pJAPPfWkWXpss3BvMDCJCw=="
+
```

Comment:
Is this Test missing the actual test code? Or is this some `gock` trickery?

Reply:
Oops, forgot. This is the point I noticed the discrepancy in the endpoint return values and went to go raise the MR in the Sia repo.